File: uapunit测试框架/bpmn/src/public/uap/workflow/bpmn2/model/EventGateway.java
package uap.workflow.bpmn2.model;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlType;
import uap.workflow.bpmn2.annotation.PropEditor;

@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name="EventGateway")
public class EventGateway extends Gateway
{
	private static final long serialVersionUID = -4229671595050383640L;
	@XmlAttribute
	public boolean instantiate;
	@XmlAttribute
	@PropEditor("uap.workflow.modeler.editors.EventGatewayTypeEditor")
	public String eventGatewayType;
	
	public boolean isInstantiate() {
		return instantiate;
	}
	public void setInstantiate(boolean instantiate) {
		this.instantiate = instantiate;
	}
	public String getEventGatewayType() {
		return eventGatewayType;
	}
	public void setEventGatewayType(String eventGatewayType) {
		this.eventGatewayType = eventGatewayType;
	}

	public String provideBeanInfoClass() {
		return "uap.workflow.modeler.bpmn2.beaninfos.EventGatewayBeanInfo";
	}
	public int getGateWayType() {
		return 3;
	}
}

File: uapunit测试框架/nchome/hotwebs/wfd/js/model/EventGateway.js
/**
 * Class: EventGateway
 */
function EventGateway(id)
{
	Gateway.call(this, id);
	this.instantiate ;
	this.eventGatewayType = "";
	this.label = "EventGateway";
};

EventGateway.prototype = new Gateway();
EventGateway.prototype.constructor = EventGateway;


/**
 * Class: EventGatewayInfo
 */
function EventGatewayInfo(id)
{
	GatewayInfo.call(this, id);
};

EventGatewayInfo.prototype = new GatewayInfo();
EventGatewayInfo.prototype.constructor = EventGatewayInfo;
